tests/amm_test.rs

```diff
@@ -627,35 +627,33 @@ mod e2e_amm {
         let bob_transfer_proof = proof_generator.read_proof(2);
         let amm_transfer_from_proof = proof_generator.read_proof(3);
 
-        info!("➡️  Sending proof for hydentity");
-        ctx.send_proof(
-            "hydentity".into(),
-            ProofData::Bytes(hydentity_proof),
-            blob_tx_hash.clone(),
-        )
-        .await?;
-
-        info!("➡️  Sending swap for amm");
-        ctx.send_proof(
-            AMM_CONTRACT_NAME.into(),
-            ProofData::Bytes(bob_swap_proof),
-            blob_tx_hash.clone(),
-        )
-        .await?;
-
-        info!("➡️  Sending transfer for hyllar");
-        ctx.send_proof(
-            "hyllar".into(),
-            ProofData::Bytes(bob_transfer_proof),
-            blob_tx_hash.clone(),
-        )
-        .await?;
+        let recursive_proof = proof_generator
+            .generate_recursive_proof(
+                &[
+                    hyle_contracts::HYDENTITY_ID,
+                    hyle_contracts::AMM_ID,
+                    hyle_contracts::HYLLAR_ID,
+                    hyle_contracts::HYLLAR_ID,
+                ],
+                &[
+                    &hydentity_proof,
+                    &bob_swap_proof,
+                    &bob_transfer_proof,
+                    &amm_transfer_from_proof,
+                ],
+            )
```

Comment:
don't you think it would be great to have both versions of the test ? One with recursion, one without ?

Reply:
(no strong opinion on this though)

Reply:
We do actually, because other parts of this test use the non-recursive setup.